### PR TITLE
feat(grid): sort groups by values count

### LINKED ISSUE #11642: Order by count in igx-grid group by
## Is your feature request related to a problem? Please describe.
When using the group by feature, you can set the sorting direction to alphabetical asc og desc, but you can sort by the number of occurrences in each group.

## Describe the solution you'd like
A new sort direction for count, or perhaps this could be default sorting if the sorting direction is set to None?
image

### PR DESCRIPTION
**Pull request by Svetloslav15**
Closes #11642

This PR fix the issue by exposing a custom sorting strategy for grouping.

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [ ] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/data-operations/sorting-strategy.ts [hunk 1]
```diff
@@ -87,4 +87,41 @@ export class DefaultSortingStrategy implements ISortingStrategy {
     protected arraySort(data: any[], compareFn?: (arg0: any, arg1: any) => number): any[] {
         return data.sort(compareFn);
     }
+}
+
+export class GroupSortingStrategy implements ISortingStrategy {
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/data-operations/sorting-strategy.ts:92]: This actually looks good to me. Rename it to `GroupMemberCountSortingStrategy`

**Review by kdinev — CHANGES_REQUESTED** [review]

**Review by kdinev — DISMISSED** [review]

**Comment by kdinev** [comment]
@Svetloslav15 Please include a `CHANGELOG.md` entry about the newly exposed grouping strategy.

**Comment by hanastasov** [comment]
THis PR needs to include UI, waiting for the design by @sdimchevski. @sdimchevski , you can run the dev demos and use the sample with the dummy UI (see comment https://github.com/IgniteUI/igniteui-angular/issues/11642#issuecomment-1460117737) to get a feeling of how this feature would work, if this helps with the UI

**Comment by kdinev** [comment]
@hanastasov We actually don't need an out-of-the-box UI for this.

**Review by kdinev — DISMISSED** [review]

**Comment by hanastasov** [comment]
> It seems that the new strategy is working as expected in almost all cases, but when I tried to use it in the "Country" column in the sample that demonstrates how the feature works, the groups weren't ordered properly:
> 
> sortingstrategyissue1 [ sortingstrategyissue1 ](https://user-images.githubusercontent.com/75325639/223952133-bf566849-02d5-4073-a3ec-1f31d48ac6c8.gif) [ ](https://user-images.githubusercontent.com/75325639/223952133-bf566849-02d5-4073-a3ec-1f31d48ac6c8.gif)

It seems the sorting logic in the new strategy needs to be enhacned, I guess it is expected to return the records sorted by the grouped column, in my testing case Country:

image

**Review by hanastasov — APPROVED** [review]

**Pull request merged**